Python2 and Python3 compatible Xena

Nandan307 opened the pull request:
Hi Falvio,
Please check your email.

Regards,
Nandan Kulkarni

fleitner commented on xenalib/XenaPort.py:
```diff
@@ -107,68 +107,68 @@ def __parse_stats(self, stats_list):
         storage = {}
         for line in stats_list:
             parms = line.split()
-            if parms[1] == 'PT_TOTAL':
```
Same comment as before, if we have a way to tell python that all strings are encoded to avoid having 'encode()' called repeatedly

Nandan307 replied:
Please give it a read : https://anonbadger.wordpress.com/2015/06/16/why-sys-setdefaultencoding-will-break-code/

fleitner commented on xenalib/XenaSocket.py:
```diff
@@ -58,6 +57,7 @@ def sendCommand(self, cmd):
         self.access_semaphor.release()
         logger.debug("sendCommand(%s) returning", cmd)
 
+    
```
Why this additional line? I am fine to remove the extra ones though :)

fleitner commented on xenalib/XenaSocket.py:
```diff
@@ -88,9 +88,11 @@ def __sendQueryReplies(self, cmd):
                 msgnew = self.bsocket.readReply()
                 msg = msgleft + msgnew
 
+
```
Same here.

fleitner commented on xenalib/XenaSocket.py:
```diff
@@ -110,6 +112,8 @@ def sendQuery(self, cmd, multilines=False):
 
         reply = self.__sendQueryReply(cmd)
         logger.debug("sendQuery(%s) reply(%s)", cmd, reply)
+
```
And here.

fleitner reviewed (changes requested):
Thank you for working on the conversion! It seems to be in the right direction but needs to address few questions before.

Nandan307 committed:
Xenalib: Conversion to run under Python2 and Python3

There was need for backwards compatible XenaPythonLib Python3 since many distributions
have been switching to default with Python3.

The main difference between Python2 and Python3 is the basic types that exist to deal
with texts and bytes. On Python3 we have one text type: "str" which holds Unicode data
and two byte types "bytes" and "bytearray".

On the other hand on Python2 we have two text types:"str" which for all intents and
purposes is limited to ASCII + some undefined data above the 7 bit range, unicode
which is equivalent to the Python3 "str" type and one byte type "bytearray" which it
inherited from Python3.

Looking at that you can see that Python3 removed the support for non Unicode data text
and in return gained a hashable byte type, the bytes "object.bytearray".

Unfortunely, Python3 doesn't provide a way to interpret all strings as a particular
type, so the conversion looks a bit ugly. Apologies.

Also much of the conversion here was provided by the "2to3" utility and everything
works fine as of now.

Signed-off-by: Nandan Kulkarni <nakulkar@redhat.com>
